src/drive/web/modules/filelist/AddFolder.jsx

```diff
@@ -1,88 +1,38 @@
 import React from 'react'
-import { connect } from 'react-redux'
-import cx from 'classnames'
+import { connect, useDispatch } from 'react-redux'
 import compose from 'lodash/flowRight'
 
 import { withClient } from 'cozy-client'
-import { translate } from 'cozy-ui/transpiled/react/providers/I18n'
-import withBreakpoints from 'cozy-ui/transpiled/react/helpers/withBreakpoints'
 import Alerter from 'cozy-ui/transpiled/react/deprecated/Alerter'
-import { TableRow, TableCell } from 'cozy-ui/transpiled/react/Table'
 
-import FilenameInput from 'drive/web/modules/filelist/FilenameInput'
 import {
   isTypingNewFolderName,
   hideNewFolderInput,
   isEncryptedFolder
 } from 'drive/web/modules/filelist/duck'
 import { createFolder } from 'drive/web/modules/navigation/duck'
-import FileThumbnail from 'drive/web/modules/filelist/FileThumbnail'
-import { Empty as EmptyCell } from 'drive/web/modules/filelist/cells'
-
-import styles from 'drive/styles/filelist.styl'
+import { AddFolderRow } from 'drive/web/modules/filelist/AddFolderRow'
 
 const AddFolder = ({
-  f,
   visible,
   isEncrypted,
   onSubmit,
   onAbort,
-  extraColumns,
-  breakpoints: { isMobile }
-}) =>
-  !visible ? null : (
-    <TableRow className={styles['fil-content-row']}>
-      <TableCell
-        className={cx(
-          styles['fil-content-cell'],
-          styles['fil-content-file-select']
-        )}
-      />
-      <TableCell
-        className={cx(
-          styles['fil-content-cell'],
-          styles['fil-file-thumbnail'],
-          {
-            'u-pl-0': !isMobile
-          }
-        )}
-      >
-        <FileThumbnail file={{ type: 'directory' }} isEncrypted={isEncrypted} />
-      </TableCell>
-      <TableCell
-        className={cx(styles['fil-content-cell'], styles['fil-content-file'])}
-      >
-        <FilenameInput onSubmit={onSubmit} onAbort={onAbort} />
-      </TableCell>
-      {!isMobile && (
-        <>
-          <TableCell
-            className={cx(
-              styles['fil-content-cell'],
-              styles['fil-content-date']
-            )}
-          >
-            <time dateTime="">{f(Date.now(), 'MMM D, YYYY')}</time>
-          </TableCell>
-          <EmptyCell className={styles['fil-content-size']} />
-          {extraColumns &&
-            extraColumns.map(column => (
-              <EmptyCell
-                key={column.label}
-                className={styles['fil-content-narrow']}
-              />
-            ))}
-          <EmptyCell className={styles['fil-content-status']} />
-        </>
-      )}
-      <TableCell
-        className={cx(
-          styles['fil-content-cell'],
-          styles['fil-content-file-action']
-        )}
+  extraColumns
+}) => {
+  if (visible) {
```

Comment:
nit: we prefer the rule `return as quick as possible` to enhance the readability 

```
if(!visible) return null
return <AddFolder...
```